ffi/src/lib.rs

```diff
@@ -1362,3 +1373,214 @@ pub unsafe extern "C" fn invoke_script(
+    /// Whether traces configuration is present
+    pub has_traces: bool,
+    /// Configuration for exporting trace data. Only valid if has_traces is true.
+    pub traces: OpenTelemetryTracesConfig,
+    /// Whether metrics configuration is present
+    pub has_metrics: bool,
+    /// Configuration for exporting metrics data. Only valid if has_metrics is true.
+    pub metrics: OpenTelemetryMetricsConfig,
+    /// Whether flush interval is specified
+    pub has_flush_interval_ms: bool,
+    /// Interval in milliseconds between consecutive exports of telemetry data. Only valid if has_flush_interval_ms is true.
+    pub flush_interval_ms: i64,
+}
+
+/// Configuration for exporting OpenTelemetry traces.
+///
+/// - `endpoint`: The endpoint to which trace data will be exported. Expected format:
+///   - For gRPC: `grpc://host:port`
+///   - For HTTP: `http://host:port` or `https://host:port`
+///   - For file exporter: `file:///absolute/path/to/folder/file.json`
+/// - `has_sample_percentage`: Whether sample percentage is specified
+/// - `sample_percentage`: The percentage of requests to sample and create a span for, used to measure command duration. Only valid if has_sample_percentage is true.
+#[repr(C)]
+#[derive(Clone)]
+pub struct OpenTelemetryTracesConfig {
+    /// The endpoint to which trace data will be exported.
+    pub endpoint: *const c_char,
+    /// Whether sample percentage is specified
+    pub has_sample_percentage: bool,
+    /// The percentage of requests to sample and create a span for, used to measure command duration. Only valid if has_sample_percentage is true.
+    pub sample_percentage: u32,
+}
+
+/// Configuration for exporting OpenTelemetry metrics.
+///
+/// - `endpoint`: The endpoint to which metrics data will be exported. Expected format:
+///   - For gRPC: `grpc://host:port`
+///   - For HTTP: `http://host:port` or `https://host:port`
+///   - For file exporter: `file:///absolute/path/to/folder/file.json`
+#[repr(C)]
+#[derive(Clone)]
+pub struct OpenTelemetryMetricsConfig {
+    /// The endpoint to which metrics data will be exported.
+    pub endpoint: *const c_char,
+}
+
+#[unsafe(no_mangle)]
+pub extern "C" fn init_open_telemetry(open_telemetry_config: OpenTelemetryConfig) -> *const c_char {
+    // At least one of traces or metrics must be provided
+    if !open_telemetry_config.has_traces && !open_telemetry_config.has_metrics {
+        let error_msg =
+            "At least one of traces or metrics must be provided for OpenTelemetry configuration";
+        return CString::new(error_msg)
+            .unwrap_or_else(|_| CString::new("Unknown error occurred").unwrap())
+            .into_raw();
+    }
+
+    let mut config = GlideOpenTelemetryConfigBuilder::default();
+
+    // Initialize open telemetry traces exporter
+    if open_telemetry_config.has_traces {
+        let traces = open_telemetry_config.traces;
+        let endpoint = unsafe { CStr::from_ptr(traces.endpoint).to_string_lossy() };
```

Comment:
This unsafe block alone is grounds to mark this function as `unsafe`. We don't know whether or not traces.endpoint is a valid pointer. That depends on what `open_telemetry_config` the user passes in. Please also add accompanying safety docs.